src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelProxyDiscoveryManager.cs

```diff
@@ -134,63 +148,44 @@ public bool HandlePartialDiscoveryComplete(IProxyDiscoveryManager proxyDiscovery
         */
         if (allDiscoverersCompleted || IsAbortRequested)
         {
-            // Reset enumerators
-            _sourceEnumerator = null;
-
-            _currentDiscoveryDataAggregator = null;
-            _currentDiscoveryEventsHandler = null;
-
-            // Dispose concurrent executors
-            UpdateParallelLevel(0);
+            _parallelOperationManager.StopAllManagers();
 
             return true;
         }
 
-        /*  Discovery is not complete.
-            Now when both.net framework and.net core projects can run in parallel
-            we should clear manager and create new one for both cases.
-            Otherwise `proxyDiscoveryManager` instance is alredy closed by now and it will give exception
-            when trying to do some operation on it.
-        */
-        EqtTrace.Verbose("ParallelProxyDiscoveryManager: HandlePartialDiscoveryComplete: Replace discovery manager. Shared: {0}, Aborted: {1}.", SharedHosts, isAborted);
-
-        RemoveManager(proxyDiscoveryManager);
-
-        proxyDiscoveryManager = CreateNewConcurrentManager();
-        var parallelEventsHandler = new ParallelDiscoveryEventsHandler(
-            _requestData,
-            proxyDiscoveryManager,
-            _currentDiscoveryEventsHandler,
-            this,
-            _currentDiscoveryDataAggregator);
-        AddManager(proxyDiscoveryManager, parallelEventsHandler);
-
-        // Second, let's attempt to trigger discovery for the next source.
-        DiscoverTestsOnConcurrentManager(proxyDiscoveryManager);
+        _parallelOperationManager.RunNextWork(proxyDiscoveryManager);
```

Comment:
Rather than managing everything here, just tell the operation manager to run next piece of work if we have any.